ooniprobe/Model/Database/Measurement.mm

```diff
@@ -96,4 +96,24 @@ -(void)deleteObject{
     [self remove];
 }
 
+- (void)getExplorerUrl:(void (^)(NSString *))measurement_url {
```

Comment:
Please:

1. refactor the callback to take two arguments, an error and a string, and pass both of them to the callback (this is much better than using a `nil` string as a sentinel for indicating error)

2. rename the callback to be `callback` (or, `cb`, or, if you don't like this naming, which however is quite well established in some arenas, call the callback something like `onMeasurementURL`)

Reply:
1. The errors will be handled directly in the function, no need to be handled outside it. Much more because this function will be called from different places.

2. I don't understand your suggestion

Reply:
1. passing `nil` to indicate failure is an anti-pattern, please don't do that!

2. I mean that `measurement_url` is not a good name for a callback.

Reply:
This function returns the `measurement_url`. If the measurement url doesn't exists return `nil`. This doesn't indicate a failure, I just care if the measurement url exists or not because I have to delete from db the json with a measurement urls